### DISCUSSION TITLE
I can't make inference with my own trained model using: ocr = PaddleOCR(use_angle_cls=True, lang='fa', show_log = False)

### DISCUSSION THREAD

**Discussion by aliiafkari** [PaddlePaddle/PaddleOCR#13705] [Q&A]
Hi everyone,

I'm encountering an issue with using my custom-trained model in PaddleOCR. Although I've successfully exported my model, the code keeps defaulting to the built-in recognition model whenever I run this cell. Does anyone know how I can switch to using my own model?

![image](https://github.com/user-attachments/assets/e5ba2673-dec8-4e5d-8a35-9079a0ea7453)

**Comment by GreatV** [comment]
To use your own trained model with PaddleOCR, you can specify the paths to your custom detection and recognition models using the `det_model_dir` and `rec_model_dir` parameters. Here's how you can modify your code:

```python
from paddleocr import PaddleOCR

# Initialize PaddleOCR with custom model directories and character dictionary
ocr = PaddleOCR(
    use_angle_cls=True, 
    lang='fa', 
    show_log=False, 
    det_model_dir='/path/to/your/detection/model/directory',
    rec_model_dir='/path/to/your/recognition/model/directory',
    rec_char_dict_path='/path/to/your/character/dictionary.txt'
)

# Perform OCR on an image
img_path = 'path/to/your/image.jpg'
result = ocr.ocr(img_path)
```

### Explanation:
- **`det_model_dir`**: The directory where your custom trained detection model is saved.
- **`rec_model_dir`**: The directory where your custom trained recognition model is saved.
- **`rec_char_dict_path`**: This parameter points to a text file containing the list of characters used in your recognition model. Each line in the file corresponds to a character that the model can recognize.

Ensure that the model files are correctly placed in the specified directories.

If you're still encountering issues, make sure the models were exported correctly and that the directory paths are accurate.
